### PR TITLE
Flesh out documentation

### PR DESCRIPTION
**Pull request by joho**
Now that Keyring has been extracted out to a fully fledged package I'm trying to bring the docs up to scratch. Still needs a fair bit of work. @lox I'd appreciate any input here that could help, I've mostly been reading the code and trying to capture the intent.

I did notice a few constants that fell on the wrong side of the aws-vault split which I'll clean up in a later PR.

### CODE REVIEW

**Comment by joho** [comment]
Probably need https://github.com/keybase/go-keychain/pull/13 merged for these docs to be true (and the lib to work properly)

**Comment by lox** [comment]
👀

**Comment by lox** [comment]
> Explain the weird TrustSelf flag.

Lol.

**Comment by lox** [comment]
Yeah, unless you use my fork at github.com/lox/go-keychain

File: README.md [hunk 1]
```diff
@@ -1,23 +1,49 @@
...
 ## Installing
 
 `go get github.com/99designs/keyring`
 
 ## Usage
 
-Something on how to use it or link to godocs
+The short version of how to use keyring is shown below.
+
+``go
```
Comment (lox) [inline README.md]: Needs an extra `

**Comment by joho** [comment]
Oh, I am already using the fork. For some reason I thought aws-vault got _off_ the fork and cos I extracted I would be on. Checking `vendor.json` sorted me out.

File: keyring.go [hunk 2]
```diff
@@ -32,20 +39,30 @@ func Open(name string, backend string) (Keyring, error) {
...
 }
 
+// Item is an, uh, item that is stored on the keyring
 type Item struct {
 	Key         string
 	Data        []byte
 	Label       string
 	Description string
-	TrustSelf   bool
+	// macOS only, set false if you want the password prompt to appear every time
```
Comment (lox) [inline keyring.go]: Reckon we should just ditch this and have a sensible default

Comment (lox) [inline keyring.go]: I guess the other option would be to have:

```go
type Item struct {
 	Key          string
 	Data         []byte
 	Label        string
 	Description  string
  KeychainOpts struct {
    TrustSelf   bool
  }
}
```

Comment (lox) [inline keyring.go]: Wow, formatting. You get my point anyway.

Comment (joho) [inline keyring.go]: I reckon the API right now isn't quite right, but probably beyond current PR scope to fix. I reckon it needs a bit more thought, I'm not quite sure what the best approach is to support functionality specific to a backend while presenting a generic interface as your API.

**Comment by joho** [comment]
I think I'll merge as is. 

It mostly explains what it does now, and has uncovered a couple of design changes to be made in future (the hard coded aws-vaults and the trustself flag) which I'll look into a little further down the line.

**Pull request merged**